lib/package.gi

```diff
@@ -159,6 +159,159 @@ InstallGlobalFunction( SetPackageInfo, function( record )
     end );
 
 
+#############################################################################
+##
+#F  FindPackageInfosInSubdirectories( pkgdir, name )
+##
+##  Finds all PackageInfos in subdirectories of directory name in
+##  directory pkgdir, return a list of their paths.
+##
+BindGlobal( "FindPackageInfosInSubdirectories", function( pkgdir, name )
+    local pkgpath, file, files, subdir;
+    
+    if name in [ ".", ".." ] then
+      return [];
+    fi;
+
+    pkgpath:= Filename( [ pkgdir ], name );
+
+    # This can be 'fail' if 'name' is a void link.
+    if pkgpath = fail then
+      return [];
+    fi;
+
+    if not IsDirectoryPath( pkgpath ) then
+      return [];
+    fi;
+
+    file:= Filename( [ pkgdir ],
+                      Concatenation( name, "/PackageInfo.g" ) );
+    if file = fail then
+      files := [];
+      # Perhaps some subdirectories contain `PackageInfo.g' files.
+      for subdir in Set( DirectoryContents( pkgpath ) ) do
+        if not subdir in [ ".", ".." ] then
+          pkgpath:= Filename( [ pkgdir ],
+                              Concatenation( name, "/", subdir ) );
+          if pkgpath <> fail and IsDirectoryPath( pkgpath )
+                              and not subdir in [ ".", ".." ] then
+            file:= Filename( [ pkgdir ],
+                Concatenation( name, "/", subdir, "/PackageInfo.g" ) );
```

Comment:
Hmm... in light of this, why is it even necessary to create and check `pkgpath`? If `pkgpath` does not refer to a dir, then surely `file` here also will be set to `fail`, no?

Reply:
No, `subdir` is just one element of the `DirectoryContents`, not really a subdir.

Reply:
I don't understand... Sure, it can be anything, e.g. a regular file. And if that happens, `.../subir/PackageInfo.g` of course is not a valid file, so `Filename` will return `fail`. What am I missing?

Reply:
Ah, I understand. No, we can remove this declaration.